// carrousel/Gruntfile.js
'use strict';

module.exports = function(grunt) {
    grunt.initConfig({
        pkg: grunt.file.readJSON('package.json'),
        name: 'carrousel',
        srcPath: 'src',
        assetsPath: 'assets',
        distPath: 'build',
        apidocPath: 'api',

        clean: ['<%= distPath %>/*'],

        copy: {
            main: {
                files: [{
                    expand: true,
                    cwd: './',
                    src: ['package.json'],
                    dest: '<%= distPath %>'
                }]
            }
        },

        depconcat: {
            options: {
                separator: '\n'
            },

            main: {
                src: ['<%= srcPath %>/<%= name %>.js'],
                dest: '<%= distPath %>/<%= name %>.debug.js'
            }
        },

        uglify: {
            main: {
                files: [{
                    expand: true,
                    cwd: '<%= distPath %>',
                    src: ['*.debug.js'],
                    dest: '<%= distPath %>',
                    ext: '.js'
                }]
            }
        },

        less: {
            options: {
                paths: ['<%= assetPath %>']
            },

            main: {
                files: [{
                    expand: true,
                    cwd: '<%= assetsPath %>',
                    src: ['<%= name %>.less'],
                    dest: '<%= distPath %>',
                    ext: '.debug.css'
                }]
            }
        },

        cssmin: {
            options: {
                report: 'min'
            },

            main: {
                files: [{
                    expand: true,
                    cwd: '<%= distPath %>',
                    src: ['*.debug.css'],
                    dest: '<%= distPath %>',
                    ext: '.css'
                }]
            }
        },

        css2js: {
            main: {
                files: {
                    '<%= distPath %>/<%= name%>_css.debug.js': '<%= distPath %>/<%= name%>.css'
                }
            }
        },

        watch: {
               
            js: {
                files: ['<%= srcPath %>/*.js', '<%= srcPath %>/**/*.js'],
                tasks: ['depconcat', 'uglify']
            },

            css: {
                files: ['<%= assetsPath %>/*.less', '<%= assetsPath %>/**/*.less'],
                tasks: ['less', 'cssmin', 'css2js']
            }

        },

        commonizor: {
            main: {
                files: {
                    '<%= distPath%>/<%= name%>.common.js': ['<%=distPath%>/<%= name %>_css.js', '<%=distPath%>/<%= name %>.js']
                }
            }
        }

    });

    grunt.loadNpmTasks('grunt-depconcat');
    grunt.loadNpmTasks('grunt-contrib-clean');
    grunt.loadNpmTasks('grunt-contrib-copy');
    grunt.loadNpmTasks('grunt-contrib-uglify');
    grunt.loadNpmTasks('grunt-contrib-less');
    grunt.loadNpmTasks('grunt-contrib-cssmin');
    grunt.loadNpmTasks('grunt-css2js');
    grunt.loadNpmTasks('grunt-contrib-watch');
    grunt.loadNpmTasks('grunt-commonizor');

    grunt.registerTask('dist', ['clean', 'less', 'cssmin', 'css2js', 'depconcat', 'uglify', 'commonizor', 'copy']);
    grunt.registerTask('dev', ['watch']);

    grunt.registerTask('default', ['dist']);
}
